src/screens/Home/stores/postStore.ts

```diff
@@ -41,21 +41,19 @@ export const PostsStore = types
   })
   .views((self) => {
     function getPostForFilter() {
-      let posts = self.posts;
-
       if (self.activeFilter === 'series') {
-        posts = filterFirstSeriesPost(
+        return filterFirstSeriesPost(
           self.posts as PostEdges,
         ) as PostsModelInstance;
       }
 
       if (self.activeFilter === 'single') {
-        posts = self.posts.filter((p) =>
-          R.isNil(p.node.frontmatter?.series),
-        ) as PostsModelInstance;
+        return self.posts
```

Comment:
For some reason, after upgrading to `mobx@5.15.5` this code stop to work so I needed to return exactly what the code expect (JS code not mobx store) in order to make it work